app/docker/networks/queries.ts

```diff
@@ -0,0 +1,79 @@
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+
+import { EnvironmentId } from '@/portainer/environments/types';
+import {
+  error as notifyError,
+  success as notifySuccess,
+} from '@/portainer/services/notifications';
+
+import { ContainerId } from '../containers/types';
+
+import {
+  getNetwork,
+  deleteNetwork,
+  disconnectContainer,
+} from './network.service';
+import { NetworkId } from './types';
+
+export function useNetwork(environmentId: EnvironmentId, networkId: NetworkId) {
+  return useQuery(
+    ['environments', environmentId, 'docker', 'networks', networkId],
+    () => getNetwork(environmentId, networkId),
+    {
+      onError: (err) => {
+        notifyError('Failure', err as Error, 'Unable to get network');
+      },
+    }
+  );
+}
+
+export function useDeleteNetwork() {
+  return useMutation(
+    ({
+      environmentId,
+      networkId,
+    }: {
+      environmentId: EnvironmentId;
+      networkId: NetworkId;
+    }) => deleteNetwork(environmentId, networkId),
+    {
+      onSuccess: (networkId) => {
+        notifySuccess('Network successfully removed', networkId);
+      },
+      onError: (err) => {
+        notifyError('Failure', err as Error, 'Unable to remove network');
+      },
+    }
+  );
+}
+
+export function useDisconnectContainer(
+  environmentId: EnvironmentId,
+  networkId: NetworkId
+) {
+  const client = useQueryClient();
+
+  return useMutation(
+    ({ containerId }: { containerId: ContainerId }) =>
```

Comment:
same as with useDelete I think you can pass envId and networkId here. You can also get them as one the parameters of OnSuccess

Reply:
updated to pass in the arguments with the .mutate function only